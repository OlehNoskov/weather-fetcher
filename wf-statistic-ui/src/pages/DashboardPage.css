.App {

    .statistic-page {
        margin-top: 40px;

        .title-statistic {
            text-align: center;
            font-size: 32px;
            font-family: sans-serif;
        }

        .statistic-diagram {
            width: 1000px;
            display: table;
            margin: 30px auto;
        }

        .buttons-group {
            display: flex;
            margin: auto;
            padding-left: 225px;

            .filter-statistic {
                margin-top: 10px;
                margin-left: 150px;
            }

            .date-statistic {
                margin-top: 10px;
                margin-left: 100px;
            }

            .reset-filter {
                margin-left: 30px;
            }
        }

        .get-statistic-country {
            text-align: center;
            margin-top: 25px;

            .search-statistic-button {
                margin-top: 5px;
                margin-left: 15px;
            }
        }
    }
}