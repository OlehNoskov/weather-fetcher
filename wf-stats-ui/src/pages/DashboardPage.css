.header {
    display: flex;
    justify-content: center;
}

.main-title {
    display: flex;
    justify-content: center;
    font-size: 40px;
    text-align: center;
    padding: 10px;
}

.inputs {
    display: flex;
    justify-content: center;
    margin-bottom: 50px;
    margin-top: 50px;

    .city {
        margin-left: 20px;
    }

    .search-button {
        margin-top: 10px;
        margin-left: 15px;
    }
}

.box {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 500px;
    background-color: #f6f6f6;
    padding: 4px;
    border-radius: 5px;

    .modal-title {
        margin-left: 10px;
        margin-top: 10px;
        font-family: sans-serif, serif;
        font-size: 20px;
    }

    .modal-close-button {
        margin-right: 10px;
        float: right;
        min-width: 0;
        width: 20px;
        height: 20px;
        line-height: 0;
    }

    .close-image {
        margin: 0;
        height: 20px;
    }

    .weather-info {
        margin-left: 30px;

        .location {
            font-size: 26px;
            font-family: -apple-system, serif;
        }

        .overall-image {
            display: inline-block;
        }

        .degrees {
            position: absolute;
            top: 130px;
            display: inline-block;
            margin-left: 80px;
            justify-content: center;
            font-size: 72px;
        }
    }
}