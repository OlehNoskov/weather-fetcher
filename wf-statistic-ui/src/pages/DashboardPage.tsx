import React from "react";
import './DashboardPage.css';
import StatisticPage from "./StatisticPage";

export default function DashboardPage() {

    return (
        <div className="App">
            <StatisticPage/>
        </div>
    );
};