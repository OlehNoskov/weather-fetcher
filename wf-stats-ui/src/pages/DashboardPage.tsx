import React, {useEffect, useState} from "react";
import {Alert, Box, Button, Icon, Modal, TextField, Typography} from "@mui/material";
import {getForecast} from "../service/Service";
import './DashboardPage.css';
import {Forecast} from "../dto/Forecast";
import StatisticPage from "./StatisticPage";
import ModalForecast from "../components/ModalForecast";

export default function DashboardPage() {
    const [forecast, setForecast] = useState<Forecast>();
    const [country, setCountry] = useState(window.localStorage.getItem("country"));
    const [city, setCity] = useState(window.localStorage.getItem("city"));
    const [open, setOpen] = React.useState(false);
    const handleOpen = () => setOpen(true);

    const getForecastWeather = () => {
        getForecast(country!, city!).then((response) => setForecast(response));
        handleOpen();
    };

    const isDisabledButton = (): boolean => {
        return country?.length === 0 || city?.length === 0;
    };

    useEffect(() => {
        window.localStorage.setItem("country", country!);
    }, [country]);

    useEffect(() => {
        window.localStorage.setItem("city", city!);
    }, [city]);

    return (
        <div className="App">
            {isDisabledButton() ?
                <Alert severity="info" className={"header"}>
                    Please, input country and city for showing weather forecast!
                </Alert>
                :
                <></>
            }
            <div className={"inputs"}>
                <div>
                    <TextField className={"input"}
                               label="Country" variant="outlined"
                               type="text" value={country}
                               onChange={(value) => {
                                   setCountry(value.target.value);
                               }} required={true}/>
                </div>
                <div>
                    <TextField label="City" variant="outlined"
                               className={"city"} type="text" value={city}
                               onChange={(value) => {
                                   setCity(value.target.value);
                               }} required={true}/>
                </div>
                <Button variant="contained" onClick={getForecastWeather}
                        className={"search-button"} disabled={isDisabledButton()}>
                    Search
                </Button>
                {/*<Modal*/}
                {/*    open={open}*/}
                {/*    onClose={handleClose}*/}
                {/*    aria-labelledby="modal-modal-title"*/}
                {/*    aria-describedby="modal-modal-description">*/}
                {/*    <Box className={"box"}>*/}
                {/*        <Typography id="modal-modal-title" variant="h6" component="h2">*/}
                {/*            <Button className="modal-close-button" onClick={changeOpenState}>*/}
                {/*                <Icon>*/}
                {/*                    <img className="close-image" src="/images/close-icon.svg" alt={'close-icon'}/>*/}
                {/*                </Icon>*/}
                {/*            </Button>*/}
                {/*            Forecast*/}
                {/*        </Typography>*/}
                {/*        <Typography id="modal-modal-description" sx={{mt: 2}}>*/}

                {/*        </Typography>*/}
                {/*    </Box>*/}
                {/*</Modal>*/}
                <ModalForecast open={open}/>
            </div>
            <div className={"statistic-page"}>
                <StatisticPage/>
            </div>
        </div>
    );
};