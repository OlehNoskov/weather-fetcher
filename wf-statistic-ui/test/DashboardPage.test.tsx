import {render, waitFor} from '@testing-library/react';
import '@testing-library/jest-dom';
import {MemoryRouter} from "react-router-dom";
import DashboardPage from "../src/pages/DashboardPage";

describe('DashboardPage', () => {
    it('should render DashboardPage without open modal and statistic diagram', async () => {
        const {asFragment} = render(
            <MemoryRouter>
                <DashboardPage/>
            </MemoryRouter>);

        await waitFor(() => {
            expect(asFragment()).toMatchSnapshot();
        });
    });
});