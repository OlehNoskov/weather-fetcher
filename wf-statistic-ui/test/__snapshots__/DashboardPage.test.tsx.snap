// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage should render DashboardPage without open modal and statistic diagram 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Loading...
    </h1>
    <div
      class="statistic-page"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 title-statistic css-2ulfj5-MuiTypography-root"
      >
        Weather statistic
      </h6>
      <div
        class="buttons-group"
      >
        <div
          class="filter-statistic"
        >
          <div
            aria-label="outlined primary button group"
            class="MuiButtonGroup-root MuiButtonGroup-contained css-1d9bq45-MuiButtonGroup-root"
            role="group"
          >
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-firstButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="/countries"
              tabindex="0"
            >
              Countries
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-middleButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="/cities"
              tabindex="0"
            >
              Cities
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-lastButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              href="/cities/country"
              tabindex="0"
            >
              Cities in country
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="date-statistic"
        >
          <div
            aria-label="outlined primary button group"
            class="MuiButtonGroup-root MuiButtonGroup-contained css-1d9bq45-MuiButtonGroup-root"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-firstButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              1 week
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-middleButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              2 weeks
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-lastButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              1 month
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            aria-label="outlined primary button group"
            class="MuiButtonGroup-root MuiButtonGroup-contained reset-filter css-1d9bq45-MuiButtonGroup-root"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1sszw6i-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Reset filter
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="statistic-diagram"
      />
    </div>
  </div>
</DocumentFragment>
`;
